opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -57,63 +96,38 @@ def set_value(
     return Context(new_values)
 
 
-def remove_value(
-    key: str, context: typing.Optional[Context] = None
-) -> Context:
-    """To remove a value, this method returns a new context with the key
-    cleared. Note that the removed value still remains present in the old
-    context.
-
-    Args:
-        key: The key of the entry to remove
-        context: The context to copy, if None, the current context is used
-    """
-    if context is None:
-        context = get_current()
-    new_values = context.copy()
-    new_values.pop(key, None)
-    return Context(new_values)
-
-
+@_load_runtime_context  # type: ignore
 def get_current() -> Context:
     """To access the context associated with program execution,
     the RuntimeContext API provides a function which takes no arguments
     and returns a RuntimeContext.
     """
+    return _RUNTIME_CONTEXT.get_current()  # type:ignore
 
-    global _RUNTIME_CONTEXT  # pylint: disable=global-statement
-    if _RUNTIME_CONTEXT is None:
-        # FIXME use a better implementation of a configuration manager to avoid having
-        # to get configuration values straight from environment variables
-        if version_info < (3, 5):
-            # contextvars are not supported in 3.4, use thread-local storage
-            default_context = "threadlocal_context"
-        else:
-            default_context = "contextvars_context"
-
-        configured_context = environ.get(
-            "OPENTELEMETRY_CONTEXT", default_context
-        )  # type: str
-        try:
-            _RUNTIME_CONTEXT = next(
-                iter_entry_points("opentelemetry_context", configured_context)
-            ).load()()
-        except Exception:  # pylint: disable=broad-except
-            logger.error("Failed to load context: %s", configured_context)
 
-    return _RUNTIME_CONTEXT.get_current()  # type:ignore
+@_load_runtime_context  # type: ignore
+def attach(context: Context) -> object:
```

Comment:
Did you try making the token type a `TypeVar`? Just curious, I don't know that it's worth the extra typing boilerplate to do so.